dokka-runners/dokka-gradle-plugin/src/main/kotlin/engine/parameters/DokkaModuleDescriptionKxs.kt

```diff
@@ -34,4 +36,20 @@ data class DokkaModuleDescriptionKxs(
     val moduleOutputDirName: String = "module",
     /** name of the sibling directory that contains the module includes */
     val moduleIncludesDirName: String = "includes",
-)
+) {
+    internal companion object {
+        fun toJsonObject(module: DokkaModuleDescriptionKxs): JsonObject = buildJsonObject {
+            put("name", module.name)
+            put("modulePath", module.modulePath)
+            put("moduleOutputDirName", module.moduleOutputDirName)
+            put("moduleIncludesDirName", module.moduleIncludesDirName)
+        }
+
+        fun fromJsonObject(obj: JsonObject): DokkaModuleDescriptionKxs = DokkaModuleDescriptionKxs(
+            name = obj["name"]!!.jsonPrimitive.content,
+            modulePath = obj["modulePath"]!!.jsonPrimitive.content,
+            moduleOutputDirName = obj["moduleOutputDirName"]!!.jsonPrimitive.content,
+            moduleIncludesDirName = obj["moduleIncludesDirName"]!!.jsonPrimitive.content,
+        )
```

Comment:
This needs to be updated, because properties with default constructor arguments are optional.

What do you think about adding some tests? To ensure any code changes don't break things in the future.